sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java

```diff
@@ -176,6 +176,13 @@ public void validate() {
             !Strings.isNullOrEmpty(this.getErrorHandling().getOutput()),
             invalidConfigMessage + "Output must not be empty if error handling specified.");
       }
+
+      if (this.getAutoSharding() != null && this.getAutoSharding()) {
```

Comment:
Wouldn't this be a check on the main BigQuery transform?

Reply:
A similar check does exist in the top-level BigQueryIO transform: https://github.com/apache/beam/blob/c82b329087342594497c6ace77632e7a358a02c7/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L3226-L3230

...along with many other warnings we put in place just because we don't want to push breaking changes. In the case of a Beam 3 type of event, we would make all these checks throw errors. With that in mind, I added a `checkArgument` when introducing the numStreams option.